5home/main.cpp

```diff
@@ -0,0 +1,101 @@
+#include <thread>
+#include <iostream>
+#include <ctime>
+#include <memory>
+#include <condition_variable>
+#include <mutex>
+ 
+using std::cout;
+using std::endl;
+ 
+const double M=8e8;
+ 
+//Shared_Work holds the work need to be done
+//part1() and part2() separates the work half-and-half
+//n1 and n2 are data items
+//chunk[64] is the separation attempt.
+//we can also separate date with alignas() 
+class Shared_Work {
+public:
+    Shared_Work():n1(1),n2(1),flag1(0),flag2(0) {}
+ 
+    void part1() {
+        for(; n1<M; n1+=n1%3);
+        flag1=1;
+        cond.notify_one();
```

Comment:
Trochę skomplikowany przykład tego false sharingu wymyśliłeś :)
Wystarczyłaby struktura z 2 polami typu int i 1 wątek inkrementuje pierwsze pole, a drugi drugie. Robią to z milion razy. Wtedy zaobserwujesz false sharing. Jak dodasz to tych pól alignas(64) to nie powinno go być.